Fixed WPML AJAX issue.

vukvukovich opened the pull request:
Was causing jobs not to be listed on non-default languages. #617

mikejolley commented:
@vukvukovich are you really suggesting this is the preferred options vs actually **looking** for $_GET/$_POST 'lang' on all requests?

This makes no sense. You support ?lang=x on some permalink structures but not all. Please do developers a favour and standardise how this works. Firing actions just for WPML seems off (the previous workaround was just as bad and I don't want either ideally).

WPML seems to have the 'all plugins need to support us' approach. In my humble opinion WPML should do more to support plugins without hacks. 

I'll merge this if we need to to get basic support in, but please consider my $_GET lang request first. It's just simpler for all parties!

mikejolley commented:
Awaiting your feedback before merge..

vukvukovich commented:
Hi Mike, sorry for waiting. WPML dev team is trying their best to standardise things by following WP codex, which is best and probably only way to achieve overall plugin compatibility. 

Regarding this case, WPML has standardised approach to permalink structures also following WP codex standards. We have short document about it here: https://wpml.org/tutorials/language-urls/

About action firing you are right, I have refactored code and removed unnecessary "lang" parameter if WPML is not present, so that action is firing properly.

Please review code and tell me if there is anything else I can help and edit.

Cheers!
